File: apps/mobile/android/app/src/main/java/com/healthcoachai/app/data/models/AuthModels.kt
package com.healthcoachai.app.data.models

import com.google.gson.annotations.SerializedName

data class AuthResponse(
    val message: String,
    val user: User,
    @SerializedName("accessToken") val accessToken: String,
    @SerializedName("refreshToken") val refreshToken: String,
    @SerializedName("expiresIn") val expiresIn: Int,
    @SerializedName("sessionId") val sessionId: String
)

data class User(
    val id: String,
    val email: String,
    @SerializedName("phoneNumber") val phoneNumber: String?,
    val role: String,
    @SerializedName("emailVerified") val emailVerified: Boolean,
    @SerializedName("phoneVerified") val phoneVerified: Boolean?
)

data class LoginRequest(
    val email: String,
    val password: String
)

data class RegisterRequest(
    val email: String,
    val password: String,
    val firstName: String?,
    val lastName: String?
)

data class OTPRequest(
    @SerializedName("phoneNumber") val phoneNumber: String
)

data class OTPVerifyRequest(
    @SerializedName("phoneNumber") val phoneNumber: String,
    val otp: String
)

data class RefreshTokenRequest(
    @SerializedName("refreshToken") val refreshToken: String
)

data class ApiError(
    val message: String,
    val statusCode: Int?
) : Exception(message)

File: apps/mobile/ios/HealthCoachAI/Models/AuthModels.swift
import Foundation

struct AuthResponse: Codable {
    let message: String
    let user: User
    let accessToken: String
    let refreshToken: String
    let expiresIn: Int
    let sessionId: String
}

struct User: Codable {
    let id: String
    let email: String
    let phoneNumber: String?
    let role: String
    let emailVerified: Bool
    let phoneVerified: Bool?
}

struct LoginRequest: Codable {
    let email: String
    let password: String
}

struct OTPRequest: Codable {
    let phoneNumber: String
}

struct OTPVerifyRequest: Codable {
    let phoneNumber: String
    let otp: String
}

struct RegisterRequest: Codable {
    let email: String
    let password: String
    let firstName: String?
    let lastName: String?
}

struct RefreshTokenRequest: Codable {
    let refreshToken: String
}

struct APIError: Codable, Error {
    let message: String
    let statusCode: Int?
}